homeassistant/components/esphome/ffmpeg_proxy.py

```diff
@@ -165,11 +176,19 @@ async def prepare(self, request: BaseRequest) -> AbstractStreamWriter | None:
                 await writer.write(chunk)
                 await writer.drain()
         finally:
-            # Close connection
-            await writer.write_eof()
+            # Clean up if the conversion hasn't changed
```

Comment:
I think that we should not clean it up . We would want to allow the devie to hit Play again and it should work.

Reply:
(we should set `proc = None` though)

Reply:
This won't be possible in general unless the device and server keep perfectly in sync. If the device pauses, the loop above also pauses. As long as the TCP connection is kept open, you can resume.

If the device breaks the TCP connection due to a restart, trying to resume from this point in the stream won't work because the media header was sent long ago.

It's possible for a device to break and resume the TCP connection while keeping the media metadata, but I'd argue this is highly unlikely in practice.

Reply:
Isn't that the default for media players? You play a song, when it finishes, it's stopped, no media is streamed and you can hit PLAY to play the same song again?

Reply:
This doesn't seem to happen regardless on the ESPHome media player. Pressing "play" again after a song has finished doesn't try to GET the URL again; it just does nothing :man_shrugging: